src/HTSPTypes.h

```diff
@@ -243,6 +269,53 @@ typedef std::map<std::string, AutoRecording> AutoRecordingsMap;
 
 } // namespace htsp
 
+class CHTSPConnection;
+
+class Subscription
+{
+
+public:
+  Subscription(CHTSPConnection &conn);
+
+  bool IsActive () const;
+  eSubsriptionState GetState(void) const;
+
+  uint32_t GetId() const;
+  uint32_t GetChannelId() const;
+  uint32_t GetWeight() const;
+  int32_t  GetSpeed() const;
+
+  void SendSubscribe(uint32_t channelId , uint32_t weight , bool force = false);
+  void SendUnSubscribe();
+  bool SendSeek(int time);
+  void SendSpeed(int32_t speed, bool force = false);
+  void SendWeight(uint32_t weight);
+
+  void ParseSubscriptionStatus(htsmsg_t *m);
```

Comment:
And at least it would be nice if I could change the subscription weight inside this class (i.e. sendweight)
Needed we we want to increase weight async..